File: 74. Search a 2D Matrix/code.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int r=matrix.size();
        int c=matrix[0].size();
        int index=-1;
        for(int i=0;i<r;i++)
        {
            if(target==matrix[i][c-1])
            {
                return true;
            }
            if(target<matrix[i][c-1])
            {
                index=i;
                break;
            }
        }
        if(index==-1)
        {
            return false;
        }
        int start=0,end=c-1;
        while(start<=end)
        {
            int mid=start+(end-start)/2;
            cout<<mid<<endl;
            if(target==matrix[index][mid])
            {
                return true;
            }
            if(target<matrix[index][mid])
            {
                end=mid-1;
            }
            else{
                start=mid+1;
            }
        }
        return false;
    }
};

File: 1704. Determine if String Halves Are Alike/code.java
class Solution {
    public boolean halvesAreAlike(String s) {
        int size=s.length();
        int c=0;
        int mid=size/2;
        for(int i=0;i<mid;i++)
        {
            if(s.charAt(i)=='a'||s.charAt(i)=='e'||s.charAt(i)=='i'||s.charAt(i)=='o'||s.charAt(i)=='u'||
               s.charAt(i)=='A'||s.charAt(i)=='E'||s.charAt(i)=='I'||s.charAt(i)=='O'||s.charAt(i)=='U')
            {
                c++;
            }
        }
        for(int i=mid;i<size;i++)
        {
            if(s.charAt(i)=='a'||s.charAt(i)=='e'||s.charAt(i)=='i'||s.charAt(i)=='o'||s.charAt(i)=='u'||
               s.charAt(i)=='A'||s.charAt(i)=='E'||s.charAt(i)=='I'||s.charAt(i)=='O'||s.charAt(i)=='U')
            {
                c--;
            }
        }
        if(c==0)
        {
            return true;
        }
        return false;
    }
}
